### ISSUE TITLE
Default CNI to enabled, even for Docker runtime

### ISSUE THREAD

**Issue by afbjorklund** [kubernetes/minikube#14760]
With the release of cri-dockerd 0.2.5, it now defaults to CNI... So it should be recommended, for _all_ runtimes and single-node.

Previously it was only enabled for runtimes using CRI, such as containerd and cri-o, or multi-node. But now, also for docker.

----

This also applies to "none" driver, which has some weird special case (breaking containerd!)

From before, the default CNI plugin is "kindnet" for KIC and multi-node and "bridge" otherwise.

To be conservative with ancient versions, it can use the dockershim change point (1.24.0-alpha.2)

It was suggested before: #6094 (to use CNI), but then it was regarded as overkill (similar to CRI)

----

* ~~https://github.com/kubernetes/minikube/issues/2207~~
* ~~https://github.com/kubernetes/minikube/issues/3567~~

Basically, rules and interfaces only applied to "others"...

Now `crictl` and `/opt/cni/bin` are needed **everywhere**.

**Comment by afbjorklund** [comment]
This also means that installing CNI needs to be added to the documentation, for "none" driver:

https://minikube.sigs.k8s.io/docs/drivers/none/

But it needed that for containerd and cri-o, anyway.
